Closing a window crashes other Zed windows

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. Open a zed window for a folder with `zed .`. 
2. Navigate to another folder and also open it with `zed .`.
3. Close one of the windows.
4. Boom!

### Environment

Zed: v0.151.0 (Zed Nightly 22a791d9c77f75ad7c77c1b131e0670c9c27b277)
OS: Linux Wayland fedora 40
Memory: 30.6 GiB
Architecture: x86_64
GPU: NVIDIA RTX A3000 Laptop GPU || NVIDIA || 555.58.02

### If applicable, add mockups / screenshots to help explain present your vision of the feature


https://github.com/user-attachments/assets/05765115-f8a3-4642-966f-c18a4a15ca5f



### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
2024-08-28T07:59:31.307244453-06:00 [INFO] ========== starting zed ==========
2024-08-28T07:59:31.424491047-06:00 [INFO] perform;
2024-08-28T07:59:31.424534165-06:00 [INFO] authenticate;
2024-08-28T07:59:31.424588481-06:00 [INFO] read_command;
2024-08-28T07:59:31.424735036-06:00 [INFO] Opening main db
2024-08-28T07:59:31.424849215-06:00 [INFO] new;
2024-08-28T07:59:31.424882669-06:00 [INFO] socket reader;
2024-08-28T07:59:31.425251536-06:00 [INFO] keep_updated;
2024-08-28T07:59:31.428274016-06:00 [INFO] Using git binary path: None
2024-08-28T07:59:31.452096309-06:00 [ERROR] No such file or directory (os error 2) about ["/home/joshuabiagio/.config/github-copilot"]
2024-08-28T07:59:31.452291497-06:00 [INFO] initializing context server client
2024-08-28T07:59:31.452372428-06:00 [INFO] Registering built-in prompt template: content_prompt
2024-08-28T07:59:31.452616817-06:00 [INFO] Registering built-in prompt template: edit_workflow
2024-08-28T07:59:31.452819861-06:00 [INFO] Registering built-in prompt template: step_resolution
2024-08-28T07:59:31.453124063-06:00 [INFO] Registering built-in prompt template: terminal_assistant_prompt
2024-08-28T07:59:31.453374557-06:00 [INFO] Prompt template overrides directory found at /home/joshuabiagio/.local/share/zed/prompt_overrides.
2024-08-28T07:59:31.454060087-06:00 [INFO] extensions updated. loading 5, reloading 0, unloading 0
2024-08-28T07:59:31.456526691-06:00 [INFO] performance metrics display disabled
2024-08-28T07:59:31.456582439-06:00 [INFO] activate is not implemented on Linux, ignoring the call
2024-08-28T07:59:31.457804798-06:00 [INFO] perform;
2024-08-28T07:59:31.457838103-06:00 [INFO] authenticate;
2024-08-28T07:59:31.457874304-06:00 [INFO] read_command;
2024-08-28T07:59:31.458188749-06:00 [INFO] socket reader;
2024-08-28T07:59:31.459404758-06:00 [INFO] set environment variables from shell:/usr/bin/fish, path:/home/joshuabiagio/venv/bin:/home/joshuabiagio/.local/bin:/home/joshuabiagio/.cargo/bin:/home/joshuabiagio/.krew/bin:/home/joshuabiagio/go/bin:/usr/local/go/bin:/home/joshuabiagio/.local/bin:/home/joshuabiagio/.cargo/bin:/home/joshuabiagio/.krew/bin:/home/joshuabiagio/go/bin:/usr/local/go/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin
2024-08-28T07:59:31.461838493-06:00 [INFO] Opening main db
2024-08-28T07:59:31.461936503-06:00 [INFO] new;
2024-08-28T07:59:31.462358953-06:00 [INFO] keep_updated;
2024-08-28T07:59:31.463669062-06:00 [INFO] set status on client 0: Authenticating
2024-08-28T07:59:31.463934001-06:00 [INFO] perform;
2024-08-28T07:59:31.463987518-06:00 [INFO] authenticate;
2024-08-28T07:59:31.464041842-06:00 [INFO] read_command;
2024-08-28T07:59:31.464274917-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:31.464327763-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:31.464371537-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:31.464400819-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:31.464491485-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:31.464529788-06:00 [INFO] socket reader;
2024-08-28T07:59:31.464566797-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:31.464625113-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:31.464660598-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:31.464696537-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:31.464724498-06:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2024-08-28T07:59:31.464770375-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:31.464820989-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:31.468246426-06:00 [INFO] new;
2024-08-28T07:59:31.468677195-06:00 [INFO] keep_updated;
2024-08-28T07:59:31.568649656-06:00 [INFO] Enabling Vulkan Portability
2024-08-28T07:59:31.568696106-06:00 [INFO] Enabling color space support
2024-08-28T07:59:31.656838316-06:00 [INFO] Adapter: "NVIDIA RTX A3000 Laptop GPU"
2024-08-28T07:59:31.656907206-06:00 [INFO] Ray tracing is supported
2024-08-28T07:59:32.065999693-06:00 [INFO] Using surface present mode MAILBOX
2024-08-28T07:59:32.06604648-06:00 [WARN] Unable to forbid exclusive full screen
2024-08-28T07:59:32.074185653-06:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-28T07:59:32.074221423-06:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-28T07:59:32.074248444-06:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2024-08-28T07:59:32.080697852-06:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "NVIDIA RTX A3000 Laptop GPU", driver_name: "NVIDIA", driver_info: "555.58.02" }
2024-08-28T07:59:32.081954954-06:00 [INFO] set status on client 199060: Connecting
2024-08-28T07:59:32.084188512-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:32.084246101-06:00 [INFO] Opening main db
2024-08-28T07:59:32.107876257-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:32.10837673-06:00 [INFO] time to first window draw: 802.004368ms
2024-08-28T07:59:32.108486857-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:32.108521521-06:00 [INFO] Node runtime install_if_needed
2024-08-28T07:59:32.108835181-06:00 [INFO] Opening main db
2024-08-28T07:59:32.164298502-06:00 [INFO] Using surface present mode MAILBOX
2024-08-28T07:59:32.164352434-06:00 [WARN] Unable to forbid exclusive full screen
2024-08-28T07:59:32.179513514-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:32.179796107-06:00 [INFO] starting language server. binary path: "/home/joshuabiagio/.local/share/zed/node/node-v22.5.1-linux-x64/bin/node", working directory: "/", args: ["/home/joshuabiagio/.local/share/zed/copilot/copilot-v0.5.0/dist/agent.js", "--stdio"]
2024-08-28T07:59:32.198877988-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T07:59:32.198923343-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:32.199203934-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T07:59:32.199233533-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:32.199342047-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T07:59:32.199374291-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:32.199403785-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:32.201418587-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T07:59:32.201462596-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:32.201490587-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:32.201597072-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:32.223003232-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:32.416846571-06:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-08-28T07:59:32.443650207-06:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-08-28T13:59:32.440Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-08-28T13:59:32.440Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-08-28T07:59:32.444963798-06:00 [INFO] Language server with id 0 sent unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "b7ca279b-b6c7-46e0-9eee-79c66854f3c9",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-08-28T07:59:32.472485258-06:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-08-28T13:59:32.449Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-08-28T13:59:32.449Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2024-08-28T07:59:32.482287171-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T07:59:32.482330535-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:32.482418816-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:32.483210239-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:32.936539821-06:00 [INFO] add connection to peer
2024-08-28T07:59:32.936640933-06:00 [INFO] add_connection;
2024-08-28T07:59:32.936720868-06:00 [INFO] waiting for server hello
2024-08-28T07:59:32.954001915-06:00 [INFO] got server hello
2024-08-28T07:59:32.954153782-06:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 502, id: 812497 })
2024-08-28T07:59:32.954352039-06:00 [INFO] set status on client 199060: Connected { peer_id: PeerId { owner_id: 502, id: 812497 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-08-28T07:59:33.933925528-06:00 [INFO] Using surface present mode MAILBOX
2024-08-28T07:59:33.934078371-06:00 [WARN] Unable to forbid exclusive full screen
2024-08-28T07:59:37.866166077-06:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2024-08-28T07:59:37.887838455-06:00 [INFO] Enabling Vulkan Portability
2024-08-28T07:59:37.887872913-06:00 [INFO] Enabling color space support
2024-08-28T07:59:37.928977985-06:00 [INFO] Adapter: "NVIDIA RTX A3000 Laptop GPU"
2024-08-28T07:59:37.929040718-06:00 [INFO] Ray tracing is supported
2024-08-28T07:59:38.205144078-06:00 [INFO] Using surface present mode MAILBOX
2024-08-28T07:59:38.205192186-06:00 [WARN] Unable to forbid exclusive full screen
2024-08-28T07:59:38.212507348-06:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-28T07:59:38.212542541-06:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-28T07:59:38.212567983-06:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2024-08-28T07:59:38.217173379-06:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "NVIDIA RTX A3000 Laptop GPU", driver_name: "NVIDIA", driver_info: "555.58.02" }
2024-08-28T07:59:38.217425917-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T07:59:38.217462188-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:38.217488067-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:38.223450623-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:38.257986666-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T07:59:38.258052639-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:38.258080417-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:38.273972841-06:00 [INFO] Using surface present mode MAILBOX
2024-08-28T07:59:38.274014412-06:00 [WARN] Unable to forbid exclusive full screen
2024-08-28T07:59:38.285594047-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:38.317366358-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T07:59:38.317433289-06:00 [INFO] starting language servers for Dockerfile: dockerfile-language-server
2024-08-28T07:59:38.317462357-06:00 [INFO] starting language server "dockerfile-language-server", path: "/home/joshuabiagio/go/src/gitlab.com/vivint/horizontals/platform/platform-base", id: 1
2024-08-28T07:59:38.31760742-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T07:59:38.317646562-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:38.31768879-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T07:59:38.317867988-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T07:59:38.317902651-06:00 [INFO] starting language servers for Dockerfile: dockerfile-language-server
2024-08-28T07:59:38.318744305-06:00 [INFO] Node runtime install_if_needed
2024-08-28T07:59:38.318821361-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T07:59:38.960509924-06:00 [INFO] Node runtime install_if_needed
2024-08-28T07:59:39.032703641-06:00 [INFO] starting language server. binary path: "/home/joshuabiagio/.local/share/zed/node/node-v22.5.1-linux-x64/bin/node", working directory: "/home/joshuabiagio/go/src/gitlab.com/vivint/horizontals/platform/platform-base", args: ["/home/joshuabiagio/.local/share/zed/extensions/work/dockerfile/node_modules/dockerfile-language-server-nodejs/bin/docker-langserver", "--stdio"]
2024-08-28T07:59:39.113570731-06:00 [WARN] unhandled capability registration: Registration { id: "bb0388d3-2a70-4d12-8a92-1f9a59323690", method: "workspace/didChangeConfiguration", register_options: Some(Object {}) }
2024-08-28T07:59:42.77870001-06:00 [ERROR] window not found
2024-08-28T07:59:42.778834055-06:00 [ERROR] window not found
2024-08-28T08:00:36.825332643-06:00 [INFO] ========== starting zed ==========
2024-08-28T08:00:36.85163658-06:00 [INFO] perform;
2024-08-28T08:00:36.851684437-06:00 [INFO] authenticate;
2024-08-28T08:00:36.85172788-06:00 [INFO] read_command;
2024-08-28T08:00:36.851925231-06:00 [INFO] Opening main db
2024-08-28T08:00:36.85206911-06:00 [INFO] new;
2024-08-28T08:00:36.852105037-06:00 [INFO] socket reader;
2024-08-28T08:00:36.85249842-06:00 [INFO] keep_updated;
2024-08-28T08:00:36.854158869-06:00 [INFO] Using git binary path: None
2024-08-28T08:00:36.8775163-06:00 [ERROR] No such file or directory (os error 2) about ["/home/joshuabiagio/.config/github-copilot"]
2024-08-28T08:00:36.877733462-06:00 [INFO] initializing context server client
2024-08-28T08:00:36.877829381-06:00 [INFO] Registering built-in prompt template: content_prompt
2024-08-28T08:00:36.878071735-06:00 [INFO] Registering built-in prompt template: edit_workflow
2024-08-28T08:00:36.878254789-06:00 [INFO] Registering built-in prompt template: step_resolution
2024-08-28T08:00:36.878533554-06:00 [INFO] Registering built-in prompt template: terminal_assistant_prompt
2024-08-28T08:00:36.878721101-06:00 [INFO] Prompt template overrides directory found at /home/joshuabiagio/.local/share/zed/prompt_overrides.
2024-08-28T08:00:36.879302961-06:00 [INFO] extensions updated. loading 5, reloading 0, unloading 0
2024-08-28T08:00:36.880090376-06:00 [INFO] set environment variables from shell:/usr/bin/fish, path:/home/joshuabiagio/venv/bin:/home/joshuabiagio/.local/bin:/home/joshuabiagio/.cargo/bin:/home/joshuabiagio/.krew/bin:/home/joshuabiagio/go/bin:/usr/local/go/bin:/home/joshuabiagio/.local/bin:/home/joshuabiagio/.cargo/bin:/home/joshuabiagio/.krew/bin:/home/joshuabiagio/go/bin:/usr/local/go/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin
2024-08-28T08:00:36.881578993-06:00 [INFO] performance metrics display disabled
2024-08-28T08:00:36.88163066-06:00 [INFO] activate is not implemented on Linux, ignoring the call
2024-08-28T08:00:36.897102603-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:36.897169724-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:36.897201492-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:36.897230679-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:36.897260353-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:36.897310357-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:36.897347261-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:36.897377016-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:36.897406078-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:36.897752105-06:00 [INFO] perform;
2024-08-28T08:00:36.897814521-06:00 [INFO] authenticate;
2024-08-28T08:00:36.897912546-06:00 [INFO] read_command;
2024-08-28T08:00:36.898278519-06:00 [INFO] socket reader;
2024-08-28T08:00:36.901975814-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:36.902036687-06:00 [INFO] Opening main db
2024-08-28T08:00:36.902337513-06:00 [INFO] new;
2024-08-28T08:00:36.902787267-06:00 [INFO] keep_updated;
2024-08-28T08:00:36.903037935-06:00 [INFO] set status on client 0: Authenticating
2024-08-28T08:00:36.903107567-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:36.9032168-06:00 [INFO] perform;
2024-08-28T08:00:36.903252464-06:00 [INFO] authenticate;
2024-08-28T08:00:36.903285718-06:00 [INFO] read_command;
2024-08-28T08:00:36.9034541-06:00 [INFO] socket reader;
2024-08-28T08:00:36.903512691-06:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2024-08-28T08:00:36.90692047-06:00 [INFO] new;
2024-08-28T08:00:36.907271874-06:00 [INFO] keep_updated;
2024-08-28T08:00:36.960462773-06:00 [INFO] Enabling Vulkan Portability
2024-08-28T08:00:36.960502816-06:00 [INFO] Enabling color space support
2024-08-28T08:00:37.043169316-06:00 [INFO] Adapter: "NVIDIA RTX A3000 Laptop GPU"
2024-08-28T08:00:37.04323803-06:00 [INFO] Ray tracing is supported
2024-08-28T08:00:37.377034327-06:00 [INFO] Using surface present mode MAILBOX
2024-08-28T08:00:37.377085963-06:00 [WARN] Unable to forbid exclusive full screen
2024-08-28T08:00:37.384519077-06:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-28T08:00:37.384556892-06:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-28T08:00:37.384584165-06:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2024-08-28T08:00:37.390359438-06:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "NVIDIA RTX A3000 Laptop GPU", driver_name: "NVIDIA", driver_info: "555.58.02" }
2024-08-28T08:00:37.390824096-06:00 [INFO] set status on client 199060: Connecting
2024-08-28T08:00:37.392608492-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:37.392702186-06:00 [INFO] Opening main db
2024-08-28T08:00:37.39367128-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:37.394491897-06:00 [INFO] Opening main db
2024-08-28T08:00:37.452887384-06:00 [INFO] time to first window draw: 627.704008ms
2024-08-28T08:00:37.453006807-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:37.453050911-06:00 [INFO] Node runtime install_if_needed
2024-08-28T08:00:37.480560296-06:00 [INFO] Using surface present mode MAILBOX
2024-08-28T08:00:37.480609702-06:00 [WARN] Unable to forbid exclusive full screen
2024-08-28T08:00:37.492368389-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:37.502873408-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:00:37.502939454-06:00 [INFO] starting language servers for Dockerfile: 
2024-08-28T08:00:37.503066624-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:00:37.50310987-06:00 [INFO] starting language servers for Dockerfile: 
2024-08-28T08:00:37.503948248-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:37.529449144-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:00:37.529540869-06:00 [INFO] starting language servers for Dockerfile: 
2024-08-28T08:00:37.530653964-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:37.530697969-06:00 [INFO] starting language server. binary path: "/home/joshuabiagio/.local/share/zed/node/node-v22.5.1-linux-x64/bin/node", working directory: "/", args: ["/home/joshuabiagio/.local/share/zed/copilot/copilot-v0.5.0/dist/agent.js", "--stdio"]
2024-08-28T08:00:37.6458738-06:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-08-28T08:00:37.799222023-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:00:37.799309583-06:00 [INFO] starting language servers for Dockerfile: dockerfile-language-server
2024-08-28T08:00:37.799347381-06:00 [INFO] starting language server "dockerfile-language-server", path: "/home/joshuabiagio/go/src/gitlab.com/vivint/horizontals/platform/platform-base", id: 1
2024-08-28T08:00:37.800174546-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:37.80021847-06:00 [INFO] Node runtime install_if_needed
2024-08-28T08:00:37.827533949-06:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-08-28T14:00:37.818Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-08-28T14:00:37.818Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-08-28T08:00:37.827878711-06:00 [INFO] Language server with id 0 sent unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "690a6df5-46ba-4460-b352-647969333869",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-08-28T08:00:37.857216709-06:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-08-28T14:00:37.832Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-08-28T14:00:37.832Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2024-08-28T08:00:38.136667231-06:00 [INFO] add connection to peer
2024-08-28T08:00:38.136806153-06:00 [INFO] add_connection;
2024-08-28T08:00:38.13691515-06:00 [INFO] waiting for server hello
2024-08-28T08:00:38.161141826-06:00 [INFO] got server hello
2024-08-28T08:00:38.16131413-06:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 502, id: 813387 })
2024-08-28T08:00:38.161566819-06:00 [INFO] set status on client 199060: Connected { peer_id: PeerId { owner_id: 502, id: 813387 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-08-28T08:00:38.324642089-06:00 [INFO] Node runtime install_if_needed
2024-08-28T08:00:38.397838345-06:00 [INFO] starting language server. binary path: "/home/joshuabiagio/.local/share/zed/node/node-v22.5.1-linux-x64/bin/node", working directory: "/home/joshuabiagio/go/src/gitlab.com/vivint/horizontals/platform/platform-base", args: ["/home/joshuabiagio/.local/share/zed/extensions/work/dockerfile/node_modules/dockerfile-language-server-nodejs/bin/docker-langserver", "--stdio"]
2024-08-28T08:00:38.471966871-06:00 [WARN] unhandled capability registration: Registration { id: "f686dec6-6104-4ead-a434-88c4d64e89d3", method: "workspace/didChangeConfiguration", register_options: Some(Object {}) }
2024-08-28T08:00:44.991822297-06:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2024-08-28T08:00:45.019403598-06:00 [INFO] Enabling Vulkan Portability
2024-08-28T08:00:45.019447207-06:00 [INFO] Enabling color space support
2024-08-28T08:00:45.048836846-06:00 [INFO] Adapter: "NVIDIA RTX A3000 Laptop GPU"
2024-08-28T08:00:45.048914423-06:00 [INFO] Ray tracing is supported
2024-08-28T08:00:45.317565995-06:00 [INFO] Using surface present mode MAILBOX
2024-08-28T08:00:45.31761589-06:00 [WARN] Unable to forbid exclusive full screen
2024-08-28T08:00:45.326376996-06:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-28T08:00:45.326415118-06:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-28T08:00:45.326442585-06:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2024-08-28T08:00:45.332404192-06:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "NVIDIA RTX A3000 Laptop GPU", driver_name: "NVIDIA", driver_info: "555.58.02" }
2024-08-28T08:00:45.332774012-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:00:45.332830185-06:00 [INFO] starting language servers for Dockerfile: dockerfile-language-server
2024-08-28T08:00:45.340858197-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:45.370196544-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:00:45.370271792-06:00 [INFO] starting language servers for Dockerfile: dockerfile-language-server
2024-08-28T08:00:45.387137335-06:00 [INFO] Using surface present mode MAILBOX
2024-08-28T08:00:45.387181255-06:00 [WARN] Unable to forbid exclusive full screen
2024-08-28T08:00:45.397661998-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:45.400792279-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:00:45.400838826-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T08:00:45.401122253-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:00:45.401165963-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T08:00:45.40625008-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:00:45.40636204-06:00 [INFO] starting language servers for Dockerfile: dockerfile-language-server
2024-08-28T08:00:45.406775273-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:00:45.406826369-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T08:00:45.40685537-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T08:00:45.408722419-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:00:47.808828579-06:00 [INFO] Using surface present mode MAILBOX
2024-08-28T08:00:47.808924092-06:00 [WARN] Unable to forbid exclusive full screen
2024-08-28T08:01:45.627521954-06:00 [INFO] ========== starting zed ==========
2024-08-28T08:01:45.650629828-06:00 [INFO] Opening main db
2024-08-28T08:01:45.65073327-06:00 [INFO] perform;
2024-08-28T08:01:45.65076987-06:00 [INFO] authenticate;
2024-08-28T08:01:45.650814156-06:00 [INFO] read_command;
2024-08-28T08:01:45.651136744-06:00 [INFO] new;
2024-08-28T08:01:45.651172589-06:00 [INFO] socket reader;
2024-08-28T08:01:45.651545947-06:00 [INFO] keep_updated;
2024-08-28T08:01:45.652986078-06:00 [INFO] Using git binary path: None
2024-08-28T08:01:45.676505588-06:00 [ERROR] No such file or directory (os error 2) about ["/home/joshuabiagio/.config/github-copilot"]
2024-08-28T08:01:45.676720128-06:00 [INFO] initializing context server client
2024-08-28T08:01:45.67679665-06:00 [INFO] Registering built-in prompt template: content_prompt
2024-08-28T08:01:45.677046335-06:00 [INFO] Registering built-in prompt template: edit_workflow
2024-08-28T08:01:45.677239059-06:00 [INFO] Registering built-in prompt template: step_resolution
2024-08-28T08:01:45.677527071-06:00 [INFO] Registering built-in prompt template: terminal_assistant_prompt
2024-08-28T08:01:45.677660817-06:00 [INFO] Prompt template overrides directory found at /home/joshuabiagio/.local/share/zed/prompt_overrides.
2024-08-28T08:01:45.678297351-06:00 [INFO] extensions updated. loading 5, reloading 0, unloading 0
2024-08-28T08:01:45.678996747-06:00 [INFO] set environment variables from shell:/usr/bin/fish, path:/home/joshuabiagio/venv/bin:/home/joshuabiagio/.local/bin:/home/joshuabiagio/.cargo/bin:/home/joshuabiagio/.krew/bin:/home/joshuabiagio/go/bin:/usr/local/go/bin:/home/joshuabiagio/.local/bin:/home/joshuabiagio/.cargo/bin:/home/joshuabiagio/.krew/bin:/home/joshuabiagio/go/bin:/usr/local/go/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin
2024-08-28T08:01:45.680572547-06:00 [INFO] performance metrics display disabled
2024-08-28T08:01:45.680624426-06:00 [INFO] activate is not implemented on Linux, ignoring the call
2024-08-28T08:01:45.680694411-06:00 [INFO] Opening main db
2024-08-28T08:01:45.691675031-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:45.691720128-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:45.691749305-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:45.691784431-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:45.691821324-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:45.692409346-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:45.692448544-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:45.692521646-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:45.692560089-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:45.692600216-06:00 [INFO] perform;
2024-08-28T08:01:45.69265774-06:00 [INFO] authenticate;
2024-08-28T08:01:45.692718175-06:00 [INFO] read_command;
2024-08-28T08:01:45.693025615-06:00 [INFO] socket reader;
2024-08-28T08:01:45.696564403-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:45.69669869-06:00 [INFO] new;
2024-08-28T08:01:45.696970591-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:45.697153499-06:00 [INFO] keep_updated;
2024-08-28T08:01:45.697407211-06:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2024-08-28T08:01:45.781749766-06:00 [INFO] Enabling Vulkan Portability
2024-08-28T08:01:45.781794279-06:00 [INFO] Enabling color space support
2024-08-28T08:01:45.856732106-06:00 [INFO] Adapter: "NVIDIA RTX A3000 Laptop GPU"
2024-08-28T08:01:45.856797557-06:00 [INFO] Ray tracing is supported
2024-08-28T08:01:46.25805207-06:00 [INFO] Using surface present mode MAILBOX
2024-08-28T08:01:46.25810277-06:00 [WARN] Unable to forbid exclusive full screen
2024-08-28T08:01:46.268309688-06:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-28T08:01:46.268353223-06:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-28T08:01:46.268381329-06:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2024-08-28T08:01:46.274249354-06:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "NVIDIA RTX A3000 Laptop GPU", driver_name: "NVIDIA", driver_info: "555.58.02" }
2024-08-28T08:01:46.275377151-06:00 [INFO] set status on client 0: Authenticating
2024-08-28T08:01:46.276283908-06:00 [INFO] perform;
2024-08-28T08:01:46.276329406-06:00 [INFO] authenticate;
2024-08-28T08:01:46.276372302-06:00 [INFO] read_command;
2024-08-28T08:01:46.276719563-06:00 [INFO] socket reader;
2024-08-28T08:01:46.277278876-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:46.277374399-06:00 [INFO] Opening main db
2024-08-28T08:01:46.278933775-06:00 [INFO] Node runtime install_if_needed
2024-08-28T08:01:46.281234984-06:00 [INFO] new;
2024-08-28T08:01:46.281851621-06:00 [INFO] keep_updated;
2024-08-28T08:01:46.296608875-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:46.296838149-06:00 [INFO] Opening main db
2024-08-28T08:01:46.351791181-06:00 [INFO] set status on client 199060: Connecting
2024-08-28T08:01:46.352409344-06:00 [INFO] time to first window draw: 725.003268ms
2024-08-28T08:01:46.35252506-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:46.381515163-06:00 [INFO] Using surface present mode MAILBOX
2024-08-28T08:01:46.381563264-06:00 [WARN] Unable to forbid exclusive full screen
2024-08-28T08:01:46.395022582-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:46.404153879-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:01:46.404204086-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T08:01:46.404439333-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:01:46.404471655-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T08:01:46.404562996-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:01:46.404597059-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T08:01:46.404625513-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T08:01:46.405363595-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:46.405418231-06:00 [INFO] starting language server. binary path: "/home/joshuabiagio/.local/share/zed/node/node-v22.5.1-linux-x64/bin/node", working directory: "/", args: ["/home/joshuabiagio/.local/share/zed/copilot/copilot-v0.5.0/dist/agent.js", "--stdio"]
2024-08-28T08:01:46.59280085-06:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-08-28T08:01:46.663134933-06:00 [INFO] Initializing default prettier with plugins {}
2024-08-28T08:01:46.663195122-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T08:01:46.663226907-06:00 [INFO] starting language servers for Shell Script: 
2024-08-28T08:01:46.664008522-06:00 [WARN] request completed with error: failed to connect to the server
2024-08-28T08:01:46.664907494-06:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-08-28T14:01:46.661Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-08-28T14:01:46.661Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-08-28T08:01:46.666423289-06:00 [INFO] Language server with id 0 sent unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "d406c639-75cb-4d17-8d14-167b2e4756e3",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-08-28T08:01:46.692630338-06:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-08-28T14:01:46.670Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-08-28T14:01:46.670Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2024-08-28T08:01:47.044708528-06:00 [INFO] add connection to peer
2024-08-28T08:01:47.044864481-06:00 [INFO] add_connection;
2024-08-28T08:01:47.044994687-06:00 [INFO] waiting for server hello
2024-08-28T08:01:47.060945805-06:00 [INFO] got server hello
2024-08-28T08:01:47.061119893-06:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 502, id: 814170 })
2024-08-28T08:01:47.061381255-06:00 [INFO] set status on client 199060: Connected { peer_id: PeerId { owner_id: 502, id: 814170 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-08-28T08:01:47.667425068-06:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-28T08:01:47.688053041-06:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-28T08:01:47.716683016-06:00 [INFO] Creating a descriptor pool for at most 256 sets
2024-08-28T08:01:47.750074608-06:00 [INFO] Creating a descriptor pool for at most 256 sets
2024-08-28T08:03:54.397549941-06:00 [ERROR] window not found
<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -167,7 +167,7 @@ members = [
     # Tooling
     #
 
-    "tooling/xtask"
+    "tooling/xtask",
 ]
 default-members = ["crates/zed"]
 
@@ -324,9 +324,9 @@ async-watch = "0.3.1"
 async_zip = { version = "0.0.17", features = ["deflate", "deflate64"] }
 base64 = "0.22"
 bitflags = "2.6.0"
-blade-graphics = { git = "https://github.com/kvark/blade", rev = "fee06c42f658b36dd9ac85444a9ee2a481383695" }
-blade-macros = { git = "https://github.com/kvark/blade", rev = "fee06c42f658b36dd9ac85444a9ee2a481383695" }
-blade-util = { git = "https://github.com/kvark/blade", rev = "fee06c42f658b36dd9ac85444a9ee2a481383695" }
+blade-graphics = { git = "https://github.com/kvark/blade", rev = "e142a3a5e678eb6a13e642ad8401b1f3aa38e969" }
+blade-macros = { git = "https://github.com/kvark/blade", rev = "e142a3a5e678eb6a13e642ad8401b1f3aa38e969" }
+blade-util = { git = "https://github.com/kvark/blade", rev = "e142a3a5e678eb6a13e642ad8401b1f3aa38e969" }
 cargo_metadata = "0.18"
 cargo_toml = "0.20"
 chrono = { version = "0.4", features = ["serde"] }

--- a/crates/gpui/src/platform/blade/blade_renderer.rs
+++ b/crates/gpui/src/platform/blade/blade_renderer.rs
@@ -336,6 +336,17 @@ impl BladePipelines {
             }),
         }
     }
+
+    fn destroy(&mut self, gpu: &gpu::Context) {
+        gpu.destroy_render_pipeline(&mut self.quads);
+        gpu.destroy_render_pipeline(&mut self.shadows);
+        gpu.destroy_render_pipeline(&mut self.path_rasterization);
+        gpu.destroy_render_pipeline(&mut self.paths);
+        gpu.destroy_render_pipeline(&mut self.underlines);
+        gpu.destroy_render_pipeline(&mut self.mono_sprites);
+        gpu.destroy_render_pipeline(&mut self.poly_sprites);
+        gpu.destroy_render_pipeline(&mut self.surfaces);
+    }
 }
 
 pub struct BladeSurfaceConfig {
@@ -438,6 +449,7 @@ impl BladeRenderer {
             self.wait_for_gpu();
             self.surface_config.transparent = transparent;
             let surface_info = self.gpu.resize(self.surface_config);
+            self.pipelines.destroy(&self.gpu);
             self.pipelines = BladePipelines::new(&self.gpu, surface_info);
             self.alpha_mode = surface_info.alpha;
         }
@@ -538,8 +550,10 @@ impl BladeRenderer {
     pub fn destroy(&mut self) {
         self.wait_for_gpu();
         self.atlas.destroy();
+        self.gpu.destroy_sampler(self.atlas_sampler);
         self.instance_belt.destroy(&self.gpu);
         self.gpu.destroy_command_encoder(&mut self.command_encoder);
+        self.pipelines.destroy(&self.gpu);
     }
 
     pub fn draw(

